hub/saml-proxy/src/integration-test/java/uk/gov/ida/integrationtest/hub/samlproxy/HubMetadataIntegrationTests.java

```diff
@@ -63,41 +65,63 @@ public void getIdpMetadataFromApi_shouldWork() throws Exception {
+
+
+        //encryption certificate
+        assertThat(getKeyName(keyDescriptors, 6)).isEqualTo(HUB_ENTITY_ID);
+        assertThat(getCertificateData(keyDescriptors, 6)).isEqualTo(TestCertificateStrings.getPrimaryPublicEncryptionCert(HUB_ENTITY_ID));
+
         assertThat(entityDescriptor.getValidUntil()).isEqualTo(DateTime.now(DateTimeZone.UTC).plusHours(1));
     }
 
-    private void thenAssertIdpMetadataDto(
-            DateTime validUntil,
-            HubIdentityProviderMetadataDto metadata
-    ) {
-        assertThat(metadata.getEntityId()).isEqualTo(HUB_ENTITY_ID);
+    private void validateKeyDescriptor(List<KeyDescriptor> keyDescriptors, int index, String issuer) {
+        validateKeyDescriptor(keyDescriptors, index, issuer, TestCertificateStrings.PUBLIC_SIGNING_CERTS.get(issuer));
+    }
+
+    private void validateKeyDescriptor(List<KeyDescriptor> keyDescriptors, int index, String issuer, String certificate) {
+        assertThat(getKeyName(keyDescriptors, index)).isEqualTo(issuer);
+        assertThat(getCertificateData(keyDescriptors, index)).isEqualTo(certificate);
+    }
 
-        assertThat(metadata.getSigningCertificates()).hasSize(2);
-        assertThat(metadata.getSigningCertificates().get(0).getIssuerId()).isEqualTo(HUB_ENTITY_ID);
-        assertThat(metadata.getSigningCertificates().get(1).getIssuerId()).isEqualTo(HUB_ENTITY_ID);
+    private String getKeyName(List<KeyDescriptor> keyDescriptors, int index) {
+        return keyDescriptors.get(index).getKeyInfo().getKeyNames().get(0).getValue();
+    }
 
-        List<String> idpIssuerIds = metadata.getIdpSigningCertificates().stream()
-                .map(Certificate::getIssuerId)
-                .collect(Collectors.toList());
+    private String getCertificateData(List<KeyDescriptor> keyDescriptors, int index) {
+        return keyDescriptors.get(index).getKeyInfo().getX509Datas().get(0).getX509Certificates().get(0).getValue();
+    }
 
-        assertThat(idpIssuerIds).containsOnly(STUB_IDP_ONE, STUB_IDP_TWO, STUB_IDP_THREE, STUB_IDP_FOUR);
+    private EntityDescriptor getEntityDescriptor(SamlDto samlDto) {
+        return new OpenSamlXMLObjectUnmarshaller<EntityDescriptor>(samlObjectParser).fromString(samlDto.getSaml());
+    }
 
-        assertThat(metadata.getValidUntil()).isEqualTo(validUntil);
+    @Test
+    public void getSpMetadataFromApi_shouldReturnTheHubFromNewMetadataAsAnSp() throws Exception {
+        SamlDto samlDto = client.target(UriBuilder.fromUri(samlProxyAppRule.getUri("/API/metadata/sp"))).request().get(SamlDto.class);
+        EntityDescriptor entityDescriptor = getEntityDescriptor(samlDto);
+
+        assertThat(entityDescriptor.getEntityID()).isEqualTo(HUB_ENTITY_ID);
+        assertThat(entityDescriptor.getSPSSODescriptor(SAMLConstants.SAML20P_NS)).isNotNull();
+        assertThat(entityDescriptor.getSPSSODescriptor(SAMLConstants.SAML20P_NS).getAssertionConsumerServices().get(0).getLocation()).isEqualTo("http://foo.com/bar");
```

Comment:
Probably worth checking getIDPSSODescriptor is null

Reply:
I diddn't actually change that test. GH is just confused.

Reply:
but I've added an assert